unity-sdk/AndroidPlugin/OptimoveUnityPlugin/src/main/java/com/optimove/unity/plugin/UnityProxy.java

```diff
@@ -1,4 +1,376 @@
+        }
+
+
+        return OptimoveInApp.InboxMessagePresentationResult.FAILED.toString();
+    }
+
+    public static boolean inAppDeleteMessageFromInbox(long messageId) {
+        try {
+
+            List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+            for (InAppInboxItem item : items) {
+                if (item.getId() == messageId) {
+                    return OptimoveInApp.getInstance().deleteMessageFromInbox(item);
+                }
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+
+        return false;
+    }
+
+    public static boolean inAppMarkInboxItemRead(long messageId) {
+        try {
+
+            List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+            for (InAppInboxItem item : items) {
+                if (item.getId() == messageId) {
+                    return OptimoveInApp.getInstance().markAsRead(item);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+
+        return false;
+    }
+
+    public static boolean inAppMarkAllInboxItemsRead() {
+        try {
+            return OptimoveInApp.getInstance().markAllInboxItemsAsRead();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public static void inAppGetInboxSummary(String guid) {
+        try {
+
+            OptimoveInApp.getInstance().getInboxSummaryAsync((InAppInboxSummary summary) -> {
+                notifyUnityOfInboxSummary(guid, summary);
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private static void notifyUnityOfInboxSummary(String guid, InAppInboxSummary summary) {
+        JSONObject res = new JSONObject();
+        boolean success = false;
+        try {
+            res.put("guid", guid);
+            if (summary != null) {
+                success = true;
+                res.put("totalCount", summary.getTotalCount());
+                res.put("unreadCount", summary.getUnreadCount());
+            }
+            res.put("success", success);
+
+            String unityMessage = res.toString();
+            UnityPlayer.UnitySendMessage("OptimoveSdkGameObject", "InvokeInboxSummaryHandler", unityMessage);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    //-- Internal / helper APIs
+
+    static void queueOrSendPushToUnity(JSONObject msg, boolean didOpenFromPush) {
+
+        try{
+            if (UnityPlayer.currentActivity == null) {
+                JSONObject pendingPush = new JSONObject();
+                pendingPush.put("pushOpened",didOpenFromPush);
+                pendingPush.put("pushMessage",msg);
+                UnityProxy.pendingPush = pendingPush;
+            } else {
+                if (didOpenFromPush){
+                    notifyUnityOfPush(msg, "Opened");
+                }else{
```

Comment:
can reduce nesting a bit by refactoring.. e.g. 
1. `notifyUnityOfPush(msg, didOpenFromPush ? "Opened" ...);`
2. reversing order, like 
```
if (UnityPlayer.currentActivity != null) {
notifyUnityOfPush...
return
}
try
```

or whichever way there is less nesting

Reply:
reduced nesting